packages/osc-ecommerce/app/root.tsx

```diff
@@ -326,7 +338,9 @@ const Document = ({ children }: DocumentProps) => {
                 <Links />
                 <DynamicLinks />
                 {/* Google Tag Manager */}
-                {process.env.NODE_ENV === 'development' || !siteSettings.analytics.gtmId ? null : (
+                {process.env.NODE_ENV === 'development' ||
+                !siteSettings.analytics.gtmId ||
+                !acceptedTos ? null : (
```

Comment:
now we have defined a new `showTracking` variable we can use it in these instances:

```suggestion
                {showTracking ? (
                    <script
                        async
                        id="gtm"
                        dangerouslySetInnerHTML={{
                            __html: `(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start': new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0], j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src= 'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f); })(window,document,'script','dataLayer','${siteSettings.analytics.gtmId}');`,
                        }}
                    />
                ) : null }
```

apply to the other 2 instances of this.

I tested it and it seemed to work based on defaulting to true here: https://github.com/Open-Study-College/osc/pull/1680/files#r1352832734

Reply:
Let me know how this goes - i can share you ID's if necessary